modules/claims_api/lib/evss_service/base.rb

```diff
@@ -109,6 +98,77 @@ def log_outcome_for_claims_api(action, status, response, claim)
         ClaimsApi::Logger.log('526_docker_container',
                               detail: "EVSS DOCKER CONTAINER #{action} #{status}: #{response}", claim: claim&.id)
       end
+
+      def error_handler(error, claim, method)
+        handle_faraday_failure(error, claim, method)
+        handle_bad_request(error, claim, method)
+        handle_parsing_error(error, claim, method)
+        handle_client_errors(error, claim, method)
+        raise error
+      end
+
+      def handle_faraday_failure(error, claim, method)
+        if error.is_a?(Faraday::ConnectionFailed)
+          status = get_status_code(error)
+          log_info = get_log_info(error, status)
+          log_outcome_for_claims_api("claims_api-526-#{method}", 'error', log_info, claim)
+          raise ::Common::Exceptions::ServiceUnavailable
+        end
+      end
+
+      def handle_client_errors(error, claim, method)
+        if ::Common::Client::Errors::ClientError || StandardError
+          status = get_status_code(error)
+          log_info = get_log_info(error, status)
+          log_outcome_for_claims_api("claims_api-526-#{method}", 'error', log_info, claim)
+
+          raise ::Common::Exceptions::Forbidden if status == 403
+          raise ::Common::Exceptions::BadRequest if [400, nil].include?(status)
+          raise ::Common::Exceptions::Authorization if status == 401
+          raise ::Common::Exceptions::ServiceError if status == 503
+        end
+      end
+
+      def handle_parsing_error(error, claim, method)
+        if error.is_a?(Faraday::ParsingError)
+          status = get_status_code(error)
+          log_info = get_log_info(error, status)
+          log_outcome_for_claims_api("claims_api-526-#{method}", 'error', log_info, claim)
+
+          raise ::Common::Exceptions::BadRequest if [400, nil].include?(status)
+          raise ::Common::Exceptions::InternalServerError if status == 500
+        end
+      end
+
+      def handle_bad_request(error, claim, method)
+        if error.is_a?(::Common::Exceptions::BadRequest) && status != 403
+          status = get_status_code(error)
+          log_info = get_log_info(error, status)
+          log_outcome_for_claims_api("claims_api-526-#{method}", 'error', log_info, claim)
+
+          raise ::Common::Exceptions::ServiceUnavailable
+        end
+      end
+
+      def get_status_code(error)
+        case error
+        when error.respond_to?(:status)
+          error.status
+        when error.respond_to?(:response_status)
+          error.response_status
+        when error.respond_to?(:status_code)
+          error.status_code
+        end
+      end
+
+      def get_log_info(error, status)
+        log_info = {}
+        message = error.respond_to?(:detailed_message) ? error.detailed_message : error.message
+        log_info['class'] = error.class if error.respond_to?(:class)
```

Comment:
@stiehlrod I tried testing this out.  What I did was fake an error response back from the docker container matching what originally caused this error in the first place.

```
[{
          "key"=> 400,
          "severity"=>"FATAL",
          "text"=>
          {"messages"=>
            [{
              "key"=>"form526.submit.establishClaim.serviceError", 
              "severity"=>"FATAL", 
              "text"=>"Claim not established. System error with BGS. GUID: 00797c5d-89d4-4da6-aab7-24b4ad0e4a4f"
            }]
          }
        }]
```

The error that then saved on the claim is in the attached screenshot.  there is a lot f noise in there but it basically looks like it is looking for a `message` key at this line 166, but the key in that error above at least is `messages`.

<img width="1072" alt="Screenshot 2024-02-16 at 2 25 45 PM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/83d65046-dbb2-41a9-8c1a-493139875009">